stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/EventHandlerImpl.kt

```diff
@@ -308,24 +310,39 @@ internal class EventHandlerImpl(
                     batch.addChannel(event.channel)
                 }
 
-                is MessageReadEvent -> {
-                    // get the channel, update reads, write the channel
-                    batch.getCurrentChannel(event.cid)?.let {
-                        val updatedChannel = it.apply {
-                            updateReads(ChannelUserRead(user = event.user, lastRead = event.createdAt))
+                // we use syncState to store the last markAllRead date for a given
+                // user since it makes more sense to write to the database once instead of N times.
+                is MarkAllReadEvent -> {
+                    event.totalUnreadCount.let(domainImpl::setTotalUnreadCount)
+                    event.unreadChannels.let(domainImpl::setChannelUnreadCount)
+
+                    // only update sync state if the incoming "mark all read" date is newer
+                    // this supports using event handler to restore mark all read state in setUser
+                    // without redundant db writes.
+                    domainImpl.repos.syncState.apply {
+                        select(event.user.id)?.let { state ->
+                            if (state.markedAllReadAt == null || state.markedAllReadAt?.before(event.createdAt) == true) {
+                                insert(state.copy(markedAllReadAt = event.createdAt))
+                            }
```

Comment:
`apply` is generally used when we're doing something _to_ the object, e.g. setting properties during initialization. If we want to open a scope into the repo here, `run` or `with` would be more fitting.

But a regular variable would work too, and is easier to read (I needed some time to figure out what we're calling on the repo inside the `apply` block):

```suggestion
                    val syncStateRepo = domainImpl.repos.syncState
                    syncStateRepo.select(event.user.id)?.let { state ->
                        if (state.markedAllReadAt == null || state.markedAllReadAt?.before(event.createdAt) == true) {
                            syncStateRepo.insert(state.copy(markedAllReadAt = event.createdAt))
```

Reply:
You're right. Don't know why I chose apply in this case.